src/mlpack/bindings/cli/cli_option.hpp

```diff
@@ -100,7 +100,11 @@ class CLIOption
     else
     {
       typename ParameterType<typename std::remove_pointer<N>::type>::type tmp;
-      data.value = boost::any(std::tuple<N, decltype(tmp)>(defaultValue, tmp));
+      if(arma::is_arma_type<N>::value || 
+          std::is_same<N, std::tuple<data::DatasetInfo, arma::mat>>::value)
+      	data.value = boost::any(std::tuple<N, decltype(tmp), size_t, size_t>(defaultValue, tmp, 0, 0));
+      else
+      	data.value = boost::any(std::tuple<N, decltype(tmp)>(defaultValue, tmp));
```

Comment:
Actually perhaps another approach would be to modify `ParameterType<T>::type` for these types to be `std::tuple<std::string, size_t, size_t>` instead of `std::tuple<std::string>`, and then a lot of the other code for handling the `std::tuple<T, typename ParameterType<T>::type>` can remain the same.